### ISSUE TITLE
ENOENT: no such file or directory Failed to parse source map

### ISSUE THREAD

**Issue by me-cooper** [blake-simpson/expo-jwt#33]
I get these errors while compiling in web:

```
WARNING in ./node_modules/expo-jwt/dist/lib/algorithms.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\lib\algorithms.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\lib\algorithms.ts'

WARNING in ./node_modules/expo-jwt/dist/lib/decoder.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\lib\decoder.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\lib\decoder.ts'

WARNING in ./node_modules/expo-jwt/dist/lib/encoder.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\lib\encoder.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\lib\encoder.ts'

WARNING in ./node_modules/expo-jwt/dist/lib/errors.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\lib\errors.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\lib\errors.ts'

WARNING in ./node_modules/expo-jwt/dist/lib/helpers.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\lib\helpers.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\lib\helpers.ts'

WARNING in ./node_modules/expo-jwt/dist/lib/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\lib\index.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\lib\index.ts'

WARNING in ./node_modules/expo-jwt/dist/lib/verifier.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\lib\verifier.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\lib\verifier.ts'

WARNING in ./node_modules/expo-jwt/dist/types/algorithms.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\types\algorithms.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\types\algorithms.ts'

WARNING in ./node_modules/expo-jwt/dist/types/claims.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\types\claims.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\username\Desktop\projectCollection\Node\DEBUG\authFlow\node_modules\expo-jwt\types\claims.ts'

web compiled with 9 warnings
```

How can I fix this? 

expo-jwt@1.6.1
expo@49.0.4

**Comment by blake-simpson** [comment]
Hi @me-cooper 

Thanks for reporting the issue. Could you please provide some more context to how/when this is occurring?

Which terminal command are you using? Is it part of a build process? Is it local or on a CI system?
Is this causing a hard error for your project, or is it just warnings?
Any info in this regard you can provide would be great :)

In general you can see that the `.ts` files are not included in the node module, only the JS ones: https://www.npmjs.com/package/expo-jwt/v/1.6.1?activeTab=code

**Comment by me-cooper** [comment]
Hey! Thank you for the quick reply. 

I just installed a basic expo project running:

`npx create-expo-app my-app`

Then I installed expo-jwt
`yarn add expo-jwt `

After that I added these lines in my code:

```javascript 
import { StatusBar } from 'expo-status-bar';
import { StyleSheet, Text, View } from 'react-native';


import JWT  from 'expo-jwt';



export default function App() {

  const key   = 'shh';
  const token = JWT.encode({ foo: 'bar' }, key);

  console.log(token);  

  return (
    <View>
      <Text>APP</Text>
      <StatusBar style="auto" />
    </View>
  );
}
```


console output: `eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJmb28iOiJiYXIifQ.YVoZ0LkWCMCnwEf7Nju2SJt_9mseJP1Q3RvCz4frGwM`

and ran `npm start` which executes `npx expo start`. I launched the web environment by typing `w` in console which should be the same as starting the app with `npx expo start --web`. When the app is launching the warnings appearing on the console.

In itself everything works but I feel very uncomfortable with the warnings. How can I fix them or can I simply rename files? But then I would have to do that for every install. So if I also reload my project from Git. 

Can you help me somehow or provide a fix? 
Would be very nice and thanks a lot for the help.

**Comment by blake-simpson** [comment]
Hi @me-cooper 

Thank you for the information!

So I have just disabled the sourcemap output from TypeScript as it was pointing to files that don't exist, as you have mentioned.

This is included in version 1.6.2 which I've just published to NPM.
Could you try to upgrade to that version please and see if the warnings no longer appear?

Thanks,
Blake

**Comment by me-cooper** [comment]
Hey! Thanks for your really very quick effort, I have updated the package to the latest version but still get the warnings when launching the webapp. 

Here is an excerpt of my `package.json` and attached the same error messages when launching the app. 

```json
{
  "name": "jwtauth",
  "version": "1.0.0",
  "main": "node_modules/expo/AppEntry.js",
  "scripts": {
    "start": "npx expo start",
    "android": "npx expo start --android",
    "ios": "npx expo start --ios",
    "web": "npx expo start --web"
  },
  "dependencies": {
    "@babel/preset-env": "^7.1.6",
    "@expo/webpack-config": "^18.0.1",
    "expo": "~48.0.18",
    "expo-jwt": "1.6.2",
    "expo-status-bar": "~1.4.4",
    "react": "18.2.0",
    "react-dom": "18.2.0",
    "react-native": "0.71.8",
    "react-native-web": "~0.18.10"
  },
  "devDependencies": {
    "@babel/core": "^7.20.0"
  },
  "private": true
}
```

As I noticed, these warnings really only come with the web version so far, not in my tests on iOS devices.


```shell
WARNING in ./node_modules/expo-jwt/dist/lib/algorithms.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\lib\algorithms.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\lib\algorithms.ts'

WARNING in ./node_modules/expo-jwt/dist/lib/decoder.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\lib\decoder.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\lib\decoder.ts'

WARNING in ./node_modules/expo-jwt/dist/lib/encoder.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\lib\encoder.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\lib\encoder.ts'

WARNING in ./node_modules/expo-jwt/dist/lib/errors.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\lib\errors.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\lib\errors.ts' 

WARNING in ./node_modules/expo-jwt/dist/lib/helpers.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\lib\helpers.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\lib\helpers.ts'

WARNING in ./node_modules/expo-jwt/dist/lib/index.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\lib\index.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\lib\index.ts'   

WARNING in ./node_modules/expo-jwt/dist/lib/verifier.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\lib\verifier.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\lib\verifier.ts'

WARNING in ./node_modules/expo-jwt/dist/types/algorithms.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\types\algorithms.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\types\algorithms.ts'

WARNING in ./node_modules/expo-jwt/dist/types/claims.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\types\claims.ts' file: Error: ENOENT: no such file or directory, open 'C:\Users\User\Desktop\projectCollection\Node\DEBUG\jwtAuth\node_modules\expo-jwt\types\claims.ts'

web compiled with 9 warnings
```

**Comment by blake-simpson** [comment]
Hi @me-cooper 

I'm not sure what happened to 1.6.2 (caching of the build, I think) but the map files were still included in the NPM module.

I've re-built it and published as 1.6.3, which should not have any map files at all:
https://www.npmjs.com/package/expo-jwt/v/1.6.3?activeTab=code

For safety you may want to locally delete the `expo-jwt` node_module before you install 1.6.3 to ensure you get the latest build too.
